### DISCUSSION TITLE
Diagrams.Net SVG is blurry when using embed images

### DISCUSSION THREAD

**Discussion by lettuceDestroyer** [typst/typst#4018] [Q&A] [answered]
I created a diagram in [Diagrams.Net](https://app.diagrams.net) and exported it as SVG with the option `Embed Images` enabled (see Image below) since HTML text can't be rendered. 

![image](https://github.com/typst/typst/assets/110594766/3bff39c3-aa89-4f47-a71f-95851988d928)

For some reason the text is really blurry in the PDF, does someone know how to fix that? Exporting all my diagrams as PNG would be an option but I'd prefer if I didn't have to do that as I use a lot of diagrams in my paper.

example Diagram:
![typst_architecture_diagram drawio](https://github.com/typst/typst/assets/110594766/ee5cb0c2-dc71-4dfe-a535-d5ecbb1d03ca)

Output:
![image](https://github.com/typst/typst/assets/110594766/414b918b-3cb7-455d-b00c-74f16286ee4b)

**Comment by LaurenzV** [comment, marked as answer]
I can't really reproduce this, this is how it looks for me when I embed it:

<img width="1512" alt="image" src="https://github.com/typst/typst/assets/47084093/a34737f4-a39e-4a76-b054-1dc2b61dc4e3">

**Reply by lettuceDestroyer** [reply]
I just noticed that it isn't an error but more how Acrobat Reader displays it. Meaning that when you have the zoom on 100% the image looks blurry but when you zoom in to 150% or more it actually is sharp. I only realised that thanks to your screenshot so thank you.

**Reply by LaurenzV** [reply]
Ah okay. Yeah, The issue might be that the text is still embedded as a picture, so maybe that's why Acrobat has issues on some Zoom levels. Another alternative you could try is to download the diagram as a PDF, convert it to SVG with some online tool, and then embed this SVG. This should hopefully preserve the text as paths.

Yeah, diagrams.net files unfortunately require these kinds of hacks...

**Reply by lettuceDestroyer** [reply]
May I ask what you normally use to create those types of diagrams to avoid those issues? 😹

**Reply by LaurenzV** [reply]
I don't really create diagrams, so I'm afraid I can't give any advice here. :p

**Reply by lettuceDestroyer** [reply]
Thank you anyway you've been incredibly helpful 😸
